dotnet/src/Agents/OpenAI/Internal/OpenAIClientFactory.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.ClientModel.Primitives;
+using System.Net.Http;
+using System.Threading;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.Http;
+using OpenAI;
+
+namespace Microsoft.SemanticKernel.Agents.OpenAI.Internal;
+
+/// <summary>
+/// Factory for creating <see cref="OpenAIClient"/>.
+/// </summary>
+internal static class OpenAIClientFactory
+{
+    /// <summary>
+    /// Avoids an exception from OpenAI Client when a custom endpoint is provided without an API key.
+    /// </summary>
+    private const string SingleSpaceKey = " ";
+
+    /// <summary>
+    /// Creates an OpenAI client based on the provided configuration.
+    /// </summary>
+    /// <param name="config">Configuration required to target a specific Open AI service</param>
+    /// <returns>An initialized Open AI client</returns>
+    public static OpenAIClient CreateClient(OpenAIServiceConfiguration config)
+    {
+        // Inspect options
+        switch (config.Type)
+        {
+            case OpenAIServiceConfiguration.OpenAIServiceType.AzureOpenAI:
+            {
+                AzureOpenAIClientOptions clientOptions = CreateAzureClientOptions(config);
+
+                if (config.Credential is not null)
+                {
+                    return new AzureOpenAIClient(config.Endpoint, config.Credential, clientOptions);
+                }
+
+                if (!string.IsNullOrEmpty(config.ApiKey))
+                {
+                    return new AzureOpenAIClient(config.Endpoint, config.ApiKey!, clientOptions);
+                }
+
+                throw new KernelException($"Unsupported configuration type: {config.Type}");
```

Comment:
The problem here is that no Credential or ApiKey was provided but the message doesn't reflect that. The message looks wrong to me because the `configType` is supported.

Reply:
**Context**: The factory methods for creating an ` OpenAIServiceConfiguration` absolutely require one or the other.  This exception is present to satisfy the function contract to return an `OpenAIClient` or throw.  As such, this exception isn't expected to be experienced by the caller, ever.

With that said, I think the messaging on line 53 and 45 are reversed (_state_ vs _type_).

I've also enhanced the message on line 45.

Updated